lib/elixir/lib/module.ex

```diff
@@ -788,7 +806,8 @@ defmodule Module do
       end
 
   """
-  def definitions_in(module, kind) do
+  @spec definitions_in(module, atom) :: [function_arity]
```

Comment:
Is atom too general? Perhaps the different kinds could have their own types and reused throughout the specs?